### DISCUSSION TITLE
How does the capillary pressure affect MOOSE converge?

### DISCUSSION THREAD

**Discussion by robo-boluo** [idaholab/moose#27600] [Q&A Modules: Porous Flow]
Greetings!

I recently engaged with the 1D injection porous flow example located in the Moose folder (modules/porous_flow/test/tests/fluidstate/theis_brineco2_nonisothermal.i). Specifically, I adjusted the length parameter from 200m to 100m within the [mesh] section and added capillary pressure . However, this alteration led to convergence issues. I'm now curious about the capillary pressure controlling convergence performance.

The changes:

![Screenshot 2024-05-10 at 12 49 23](https://github.com/idaholab/moose/assets/59897601/5ab39545-ce1d-4578-bc1a-49b6032f58d0)

`  [pc]
    # type = PorousFlowCapillaryPressureConst
    # pc = 0
    type = PorousFlowCapillaryPressureVG
    alpha = 5.099e-5
    m = 0.457
    sat_lr = 0.0
    pc_max = 1e7
  []`

Results:
![Screenshot 2024-05-10 at 12 52 03](https://github.com/idaholab/moose/assets/59897601/88d5b96e-88e1-4f48-9e75-daf5a09c213a)

Thank you,

**Comment by GiudGiud** [comment]
Hello

There's a page on porous flow converge issues if you want to check that out
https://mooseframework.inl.gov/modules/porous_flow/tutorial_02.html
